# Publish all joint states -----------------------------------
controller_manager:
  ros__parameters:
    update_rate: 10  # Hz

    joint_state_broadcaster:
      type: joint_state_broadcaster/JointStateBroadcaster



roboteq_controller:
  ros__parameters:
    type: "diff_drive_controller/DiffDriveController"
    left_wheel_names: ["left_joint"]
    right_wheel_names: ["right_joint"]
    publish_rate: 50.0
    pose_covariance_diagonal: [0.001, 0.001, 0.001, 0.001, 0.001, 0.03]
    twist_covariance_diagonal: [0.001, 0.001, 0.001, 0.001, 0.001, 0.03]
    cmd_vel_timeout: 0.25
    use_stamped_vel: false

    # Base frame_id
    base_frame_id: base_footprint

    # Odometry fused with IMU is published by robot_localization, so
    # no need to publish a TF based on encoders alone.
    enable_odom_tf: false

    # Panther hardware provides wheel velocities
    estimate_velocity_from_position: false

    # Wheel separation and radius multipliers
    wheel_separation:  1.5095 # default: 1.0
    wheel_radius: 0.35435
    wheel_radius_multiplier:     0.95 # default: 1.0

    # Velocity and acceleration limits
    # Whenever a min_* is unspecified, default to -max_*
    linear:
      x:
        has_velocity_limits:     true
        max_velocity:            1.0   # m/s
        has_acceleration_limits: true
        max_acceleration:        0.5   # m/s^2
    angular:
      z:
        has_velocity_limits:     true
        max_velocity:            1.0   # rad/s
        has_acceleration_limits: true
        max_acceleration:        0.5   # rad/s^2
